src/main/java/bio/terra/service/load/LoadDao.java

```diff
@@ -0,0 +1,128 @@
+package bio.terra.service.load;
+
+
+import bio.terra.common.DaoKeyHolder;
+import bio.terra.service.load.exception.LoadLockFailureException;
+import bio.terra.service.load.exception.LoadLockedException;
+import org.apache.commons.codec.binary.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.UUID;
+
+@Repository
+public class LoadDao {
+    private final Logger logger = LoggerFactory.getLogger("bio.terra.service.snapshot.SnapshotDao");
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public LoadDao(NamedParameterJdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
```

Comment:
Not up-to-date on my Spring... what does this annotation do? Does it protect against multiple calls to `lockLoad` running concurrently?

Reply:
The Transaction annotation affects the SQL session used in JDBC template. The default is auto-commit, so each SQL statement is a transaction. This annotation turns off auto-commit and makes the transaction boundaries explicit and it will commit the transaction on normal return from the method. The propagation flag means that the same transaction is used in methods called by this method.